frame/proxy/src/lib.rs

```diff
@@ -558,7 +567,13 @@ pub mod pallet {
 		_,
 		Twox64Concat,
 		T::AccountId,
-		(Vec<ProxyDefinition<T::AccountId, T::ProxyType, T::BlockNumber>>, BalanceOf<T>),
+		(
+			BoundedVec<
```

Comment:
due to the usage of bounded vec here we should make sure that the limit is not exceeded before migration.
Also any modification of `T::MaxProxies` should ensure the on chain items don't exceed the limit.

If these consideration are too strong, we should move to WeakBoundedVec.